### PR TITLE
feat(ctb): Promote FP contracts to first major version

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Promotes the fault proof contracts that are semantically versioned to
`1.0.0`, in preparation for the upcoming deployment.

### CODE REVIEW

**Comment by clabby** [comment]
* **#10382** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10382?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol
Comment (mds1) [inline packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol:1]: So is this the full set of contracts that we'll be enumerating as released for the tag release notes? Should anything in the `src/cannon` folder get bumped to 1.0.0 also?

Comment (ajsutton) [reply packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol:1]: Probably is worth making `MIPS.sol` and `PreimageOracle` be `1.0.0` as well unless there's a reason not to.

Comment (ajsutton) [reply packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol:1]: oh we have now. :)

**Review by Inphi — APPROVED** [review]

**Comment by Inphi** [comment]
@ajsutton This should be forwards compatible with op-challenger's contract versioning logic?

**Comment by ajsutton** [comment]
Edit: Sorry first version of this comment was based on me thinking this was a different PR.

But the answer is still yes this will work.  The contract versioning logic uses the latest bindings by default and checks for a version starting with `0.8` to fall back to the older compatibility bindings.

File: packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol
Comment (mds1) [reply packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol:1]: Perfect, thank you!

**Review by mds1 — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**